module/VuFindTheme/src/VuFindTheme/ColorSchemeManager.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+/**
+ * Color Scheme Manager
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Theme
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+
+namespace VuFindTheme;
+
+use Laminas\Stdlib\RequestInterface as Request;
+use Psr\Container\ContainerInterface;
+use VuFind\Config\Config;
+use VuFind\Cookie\CookieManager;
+
+use function in_array;
+
+/**
+ * Color Scheme Manager
+ *
+ * @category VuFind
+ * @package  Theme
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+class ColorSchemeManager
+{
+    /**
+     * Supported color schemes for the current theme
+     *
+     * @var array
+     */
+    protected $supportedColorSchemes;
+
+    /**
+     * Constructor
+     *
+     * @param Config             $config         Theme configuration object
+     * @param ContainerInterface $serviceManager Top-level service container
+     * @param CookieManager      $cookieManager  Cookie manager
+     * @param string             $currentTheme   Current theme
+     */
+    public function __construct(
+        protected Config $config,
+        protected ContainerInterface $serviceManager,
+        protected CookieManager $cookieManager,
+        protected string $currentTheme,
+    ) {
+        $rawSupportedColorSchemes = $this->config->supported_color_schemes[$this->currentTheme] ?? '';
```

Comment:
I don't think you need to use `$this` for these since you have access to the constructor arguments here  (unless property promotion has a side effect I'm unaware of). Might be worth making the code more concise.

```suggestion
        $rawSupportedColorSchemes = $config->supported_color_schemes[$currentTheme] ?? '';
```